intArr lectures
int32 twist
